source/m2mfirmware.cpp

```diff
@@ -293,23 +288,27 @@ void M2MFirmware::create_mandatory_resources()
     // todo:
     // perhaps we should have a API for batch creation of objects by using a array
     // of lwm2m_parameters.
-
+    firmware_package_params.identifier.name = (char*)FIRMWARE_PACKAGE;
```

Comment:
Isn't the firmware_package_params a const struct, possibly also in ROM? I guess the instance_id could be just set to a numeric value there to avoid having a "0" -string completely.